autotest/gcore/tiff_write.py

```diff
@@ -5350,6 +5350,31 @@ def test_tiff_write_123():
     gdaltest.tiff_drv.Delete("/vsimem/tiff_write_123_rgba_to_undefined.tif")
 
 
+###############################################################################
+# Test implicit photometric interpretation
+
+
+def test_tiff_write_RGBNir(tmp_vsimem):
+
+    out_filename = tmp_vsimem / "test_tiff_write_RGBNir.tif"
+    src_ds = gdaltest.tiff_drv.Create(out_filename, 1, 1, 4, gdal.GDT_Int16)
+    src_ds.GetRasterBand(1).SetColorInterpretation(gdal.GCI_RedBand)
+    src_ds.GetRasterBand(2).SetColorInterpretation(gdal.GCI_GreenBand)
+    src_ds.GetRasterBand(3).SetColorInterpretation(gdal.GCI_BlueBand)
+    src_ds.GetRasterBand(4).SetColorInterpretation(gdal.GCI_NIRBand)
+    src_ds = None
+    statBuf = gdal.VSIStatL(
+        str(out_filename) + ".aux.xml",
+        gdal.VSI_STAT_EXISTS_FLAG | gdal.VSI_STAT_NATURE_FLAG | gdal.VSI_STAT_SIZE_FLAG,
+    )
+    assert statBuf is None, "did not expect PAM file"
+    src_ds = gdal.Open(out_filename)
+    assert src_ds.GetRasterBand(1).GetColorInterpretation() == gdal.GCI_RedBand
+    assert src_ds.GetRasterBand(2).GetColorInterpretation() == gdal.GCI_GreenBand
+    assert src_ds.GetRasterBand(3).GetColorInterpretation() == gdal.GCI_BlueBand
+    assert src_ds.GetRasterBand(4).GetColorInterpretation() == gdal.GCI_NIRBand
```

Comment:
The underlying problem was that this configuration raised a warning about ExtraSamples when initially closing the file, and when subsequently opening it. Does autotest treat warnings as errors ?